packages/apps/reputation-oracle/server/src/modules/staking/staking.service.spec.ts

```diff
@@ -0,0 +1,226 @@
+import { faker } from '@faker-js/faker';
+import { createMock } from '@golevelup/ts-jest';
+import { Test, TestingModule } from '@nestjs/testing';
+import { ethers } from 'ethers';
+
+import { SupportedExchange } from '@/common/constants';
+import { StakingConfigService, Web3ConfigService } from '@/config';
+import { ExchangeClientFactory } from '@/modules/exchange/exchange-client.factory';
+import {
+  ExchangeApiKeyNotFoundError,
+  ExchangeApiKeysService,
+} from '@/modules/exchange-api-keys';
+import { UserEntity, UserNotFoundError, UserRepository } from '@/modules/user';
+import { WalletWithProvider, Web3Service } from '@/modules/web3';
+import { mockWeb3ConfigService } from '@/modules/web3/fixtures';
+
+import { StakingService } from './staking.service';
+import { ExchangeClient } from '../exchange/types';
+
+jest.mock('@human-protocol/sdk', () => {
+  const actual = jest.requireActual('@human-protocol/sdk');
+  return {
+    ...actual,
+    StakingClient: {
+      build: jest.fn(),
+    },
+  };
+});
+
+const { StakingClient } = jest.requireMock('@human-protocol/sdk');
```

Comment:
Since you already mocked it - you can import it directly as
```
import { StakingClient } from '@human-protocol/sdk';
```
and then below
```
const mockedStakingClient = jest.mocked(StakingClient);
```
so you can use typed `mockedStakingClient` everywhere you need to mock staking client behavior

You can check for examples in some other modules we have